src/win/util.c

```diff
@@ -1153,3 +1155,100 @@ int uv_getrusage(uv_rusage_t *uv_rusage) {
 
   return 0;
 }
+
+
+int uv_os_homedir(char* buffer, size_t* size) {
+  wchar_t* path;
+  char* buf;
+  size_t bufsize;
+  int r;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return UV_EINVAL;
+
+  /* Check if the USERPROFILE environment variable is set first */
+  bufsize = GetEnvironmentVariableW(L"USERPROFILE", NULL, 0);
+
+  if (bufsize == 0)
```

Comment:
Probably unlikely, but we should check GetLastError here, in case it's ERROR_ENVVAR_NOT_FOUND it's not an error, we should just continue. I see you handle it later, but I guess we need to check both times, because threads!